### DISCUSSION TITLE
"kill_to_line_end" only bound in insert mode: is this by design?

### DISCUSSION THREAD

**Discussion by jeanas** [helix-editor/helix#11956] [General] [answered]
In the [keymap](https://docs.helix-editor.com/keymap.html), I see that the (to me very useful) command `kill_to_line_end` has a key bound by default in insert mode (<kbd>Ctrl</kbd> <kbd>k</kbd>), but there is no key in normal mode. Yet it seems the insert mode bindings are discouraged: “💡 New users are strongly encouraged to learn the modal editing paradigm to get the smoothest experience.”

Is it intentional that this binding has no counterpart in normal mode?

**Comment by the-mikedavis** [comment, marked as answer]
Normal mode encourages making selections and then acting on them, so you would use something like `t<ret>` to select until the end of the line and `d` instead of a single action like `kill_to_line_end`

**Reply by jeanas** [reply]
OK, thanks. I use this often so I'll just bind <kbd>k</kbd> to `kill_to_line_end` in my config.

**Reply by markstos** [reply]
I have bound `kill_to_end_of_line` to `D` in normal mode, but I appreciate the overall consistent design of the Helix bindings and don't mind that it's not bound by default.

Also good to discover the `t<ret>d` shortcut. I had been using `vgld` and this saves a character!
